plugin/core/types.py

```diff
@@ -197,6 +282,67 @@ def __init__(self,
         self.experimental_capabilities = experimental_capabilities
         self.status_key = "lsp_{}".format(self.name)
 
+    @classmethod
+    def from_sublime_settings(cls, name: str, s: sublime.Settings, file: str) -> "ClientConfig":
+        base = sublime.decode_value(sublime.load_resource(file))
+        settings = DottedDict(base.get("settings", {}))  # defined by the plugin author
+        settings.update(read_dict_setting(s, "settings", {}))  # overrides from the user
+        return ClientConfig(
+            name=name,
+            binary_args=read_list_setting(s, "command", []),
+            languages=_read_language_configs(s),
+            tcp_port=s.get("tcp_port", None),
+            # Default to True, because an LSP plugin is enabled iff it is enabled as a Sublime package.
+            enabled=bool(s.get("enabled", True)),
+            init_options=s.get("initializationOptions"),  # type: ignore
+            settings=settings,
+            env=read_dict_setting(s, "env", {}),
+            tcp_host=s.get("tcp_host", None),
+            tcp_mode=s.get("tcp_mode", None),
+            experimental_capabilities=s.get("experimental_capabilities", None)  # type: ignore
+        )
+
+    @classmethod
+    def from_dict(cls, name: str, d: Dict[str, Any]) -> "ClientConfig":
+        return ClientConfig(
+            name=name,
+            binary_args=d.get("command", []),
+            languages=_read_language_configs(d),
+            tcp_port=d.get("tcp_port", None),
+            enabled=d.get("enabled", False),
+            init_options=d.get("initializationOptions", dict()),
```

Comment:
Would switching `init_options` to `DottedDict` be in scope of this change?
LSP-vue settings are provided through it and it's annoying that you can't use the dotted notation for it.

Reply:
Thanks. :)